### DISCUSSION TITLE
does it work with cpu too?

### DISCUSSION THREAD

**Discussion by Amin456789** [lshqqytiger/stable-diffusion-webui-directml#309] [Q&A] [answered]
hello, i would like to use it but my amd gpu is low in vram [2gb] so i was thinking to use it with cpu
is it using torch-directml or onnx?
can i use safenensors models or i have to convert to onnx?

**Comment by KanonXP** [comment]
Hello, how are you? What graphics card are you using? I'm using an ASRock RX 550 with 2GB of VRAM, 16GB of RAM, an i5 2400, and I'm using version 1.6.0 of automatic 1111 by lshqqytiger/stable-diffusion-webui-directml, with a resolution of 512px width x 712px height, 20 steps, and the Sampling method: DPM++ 2M Karras. I'm getting results like | 20/20 [02:28<00:00, 7.45s/it].

**Comment by KanonXP** [comment]
Look Amin456789, I'm sharing an image I created with the following settings, model: anything-v4.5-pruned-fp16.ckpt [f773383dbc], Sampling steps: 20, Sampling method: DPM++ 2M Karras, resolution of 512px width x 712px height, Seed: 1111729883, prompt: 1girl, Negative Prompt: badhandv4 easynegative verybadimagenegative_v1.3, CFG Scale: 7, obtaining: | 20/20 [02:28<00:00, 7.45s/it].

![00056-1111729883](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/88633126/e9fe33f5-2852-4631-9caf-768799ee585a)

@Amin456789

**Reply by Amin456789** [reply]
hello, i am good thank u, how are u?
wow that is great i didn't know it can do that. my graphic card is amd r5 m330 with 2gb vram, that speed u r saying is great!
so u r using a ckpt rather than a onnx file right? ur gpu is much stronger than mine but still great that it works with gpu 2gb ram too. what settings and command-args are u using, will be great if u share them, thank u

**Comment by KanonXP** [comment, marked as answer]
@Amin456789, Good morning, I'd be happy to share with you my arguments and configurations that I use:

@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --autolaunch --lowvram --upcast-sampling --disable-nan-check --ckpt-dir 'F:\stable-diffusion-webui-directml-master\models\Stable-diffusion' --lora-dir 'F:\stable-diffusion-webui-directml-master\models\Lora\' --embeddings-dir 'F:\stable-diffusion-webui-directml-master\embeddings'

call webui.bat

I'll also share my settings for the automatic 1111: in Settings, Optimizations.

I also recommend disabling hardware acceleration in the browser you're using and any other programs that might be using your GPU's VRAM or graphic processing.

I also recommend using FP16 models. Typically, these models will have a weight of 1.98GB and occupy less VRAM.


![Captura de pantalla 2023-11-01 104051](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/88633126/479b21f6-4e1c-4e11-86e0-65c7b5e7baee)

If you have limited RAM, I recommend using virtual memory, preferably from an SSD. However, if you don't have one, you can use it from an HDD.

**Reply by Amin456789** [reply]
good morning friend! thank u so much for ur time to do this... hope u have a great day! im gonna do it right now! thanks again

**Comment by KanonXP** [comment]
Hello @Amin456789, I hope this is useful. Feel free to reach out if you need anything; I'll be happy to assist you. I hope you share your tests; I'll be glad to see that it worked for you. I also want to share another tip with you. If you notice that your images lack detail or appear blurry, and you can't increase the resolution, you can use the Ultimate SD Upscale extension in the img2img section with the following configuration: Sampling steps: 20, Sampling method: DPM++ 2M Karras, Denoising strength: 0.5, Target size type: Scale from image size: 1.5, Upscaler: ESRGAN_4x,
![2,Captura web_1-11-2023_11179_127 0 0 1](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/88633126/d4ec7b10-6403-4c2e-8152-beb478fd397c)
If you feel it's a bit slow, you can lower the Denoising strength from 0.5 to 0.2, but you will lose some quality.

![github2](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/88633126/5d714332-8c24-494d-88fb-18464619f214)

**Reply by Amin456789** [reply]
thank u friend!

**Comment by Amin456789** [comment]
thank u for ur time friend, but it didn't work for me, it seems id doesn't use my amd gpu [im using laptop with 2 gpus] i tried many thing i saw on internet but it didn't work.  it could be driver issue as its older than maybe ur gpu or something else
im taking a bit rest from new ways in AI for now and use what i got, im putting lots of time in it :D

thanks again friend
